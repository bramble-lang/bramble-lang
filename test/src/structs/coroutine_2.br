fn my_main() -> i64 {
    let ms:MyStruct := MyStruct {x: 5, y: 13, z: true};
    let ms2:MyStruct := MyStruct {x: 123, y: 456, z: false};
    
    let c: co MyStruct := init my_co(ms, ms2);
    
    let w : MyStruct := yield (c);
    project::std::io::writei64ln(w.x);
    project::std::io::writei64ln(w.y);
    project::std::io::writeboolln(w.z);
    
    let w2 : MyStruct := yield c;
    project::std::io::writei64ln(w2.x);
    project::std::io::writei64ln(w2.y);
    project::std::io::writeboolln(w2.z);
    
    let w3 : MyStruct := yield c;
    project::std::io::writei64ln(w3.x);
    project::std::io::writei64ln(w3.y);
    project::std::io::writeboolln(w3.z);
    
    return 0;
}

co my_co (x: MyStruct, y: MyStruct) -> MyStruct {
    project::std::io::writei64ln(x.x);
    yret x;
    project::std::io::writeboolln(y.z);
    return y;
}

struct MyStruct {
    x: i64,
    y: i64,
    z: bool,
}