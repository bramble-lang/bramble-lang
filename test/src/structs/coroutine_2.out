5
5
13
true
false
123
456
false
123
456
false
